sbpack/noncwl/utils.py

```diff
@@ -279,6 +305,63 @@ def get_entrypoint(path):
         return None
 
 
+# Nextflow
+def get_latest_sb_schema(path):
+    """
+    Auto find sb_nextflow_schema file.
+    """
+    possible_paths = []
+    for file in os.listdir(path):
+        result = re.findall(
+            fr"(.*{SB_SCHEMA_DEFAULT_NAME}\.?(\d+)?\.(ya?ml|json))", file
+        )
+        if result:
+            result = result.pop(0)
+            if not result[1]:
+                prep = 0, result[0]
+            else:
+                prep = int(result[1]), result[0]
+            possible_paths.append(prep)
+
+    if possible_paths:
+        latest = sorted(possible_paths).pop()[1]
+        sb_schema_path = os.path.join(path, latest)
+        print(f"Located latest sb_nextflow_schema at <{sb_schema_path}>")
+        return sb_schema_path
+    else:
+        return None
+
+
+def get_executor_version(string):
+    result = re.findall(
+        r"\[Nextflow]\([^(]+(%E2%89%A5|%E2%89%A4|=|>|<)(\d{2}\.\d+\.\d+)[^)]+\)",
+        string
+    )
+    if result:
+        sign, version = result.pop(0)
```

Comment:
is unpacking here ok? 4 groups are expected in regular expression?

Reply:
Sorry, maybe it is ok. If this is example for executor version string: "[Nextflow]([=12.2.4])" ?